### PR TITLE
PIC-2163 map aliases correctly to CCS grouped offender matches

### PR DESCRIPTION
**Pull request by rschallapalli**

### CODE REVIEW

File: src/main/java/uk/gov/justice/probation/courtcasematcher/restclient/model/courtcaseservice/CCSMatchIdentifiers.java [hunk 1]
```diff
@@ -18,12 +21,20 @@ public class CCSMatchIdentifiers {
...
                 .pnc(matchIdentifiers.getPnc())
                 .cro(matchIdentifiers.getCro())
+                .aliases(mapAliases(matchIdentifiers))
                 .build();
     }
+
+    private static List<CCSOffenderAlias> mapAliases(MatchIdentifiers matchIdentifiers) {
+        return Optional.ofNullable(matchIdentifiers.getAliases()).map(offenderAliases ->
+                        offenderAliases.stream().map(CCSOffenderAlias::of).collect(Collectors.toList()))
+                .orElse(null);
```
Comment (vb-moj) [inline src/main/java/uk/gov/justice/probation/courtcasematcher/restclient/model/courtcaseservice/CCSMatchIdentifiers.java:38]: Could this be returns an Optional.empty() instead?

File: src/main/java/uk/gov/justice/probation/courtcasematcher/restclient/model/courtcaseservice/CCSOffenderAlias.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+@Builder
+@NoArgsConstructor(access = AccessLevel.PRIVATE, force = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class CCSOffenderAlias {
+
+    private final LocalDate dateOfBirth;
+    private final String firstName;
+    private final List<String> middleNames;
+    private final String surname;
+    private final String gender;
```
Comment (vb-moj) [inline src/main/java/uk/gov/justice/probation/courtcasematcher/restclient/model/courtcaseservice/CCSOffenderAlias.java:25]: if possible, could this be an enum?

Comment (rschallapalli) [reply src/main/java/uk/gov/justice/probation/courtcasematcher/restclient/model/courtcaseservice/CCSOffenderAlias.java:25]: John and myself had this discussion and made a decision to leave this as string. We do not own this data and is merely used as a cached copy and saved as is as JSON in the DB

File: src/main/java/uk/gov/justice/probation/courtcasematcher/restclient/model/courtcaseservice/CCSMatchIdentifiers.java (hunk 1 above)
Comment (rschallapalli) [reply src/main/java/uk/gov/justice/probation/courtcasematcher/restclient/model/courtcaseservice/CCSMatchIdentifiers.java:38]: It is the usual way but doesn't apply to the very reason I created this utility method.. the calling code isn't expecting or wanting an Optional

**Review by vb-moj — APPROVED** [review]

**Review by jhevans — APPROVED** [review]

**Pull request merged**